sbi/utils/user_input_checks.py

```diff
@@ -553,11 +557,10 @@ def validate_theta_and_x(theta: Any, x: Any) -> None:
 
     # I did not fuse these asserts with the `isinstance(x, Tensor)` asserts in order
     # to give more explicit errors.
-    assert isinstance(theta, torch.FloatTensor), "Type of parameters must be float32."
-    assert isinstance(
-        x, torch.FloatTensor
-    ), "Type of simulator outputs must be float32."
+    assert theta.dtype == float32, "Type of parameters must be float32."
+    assert x.dtype == float32, "Type of simulator outputs must be float32."
 
+    return True
```

Comment:
Why is this needed?

Reply:
this way, the validation function could be used to infer the state explicitely. the current usage is a bit `fortran`-like for my taste. Feel free to disagree. I can remove that in no time.